dangerfile.ts

```diff
@@ -53,3 +54,15 @@ if (fs.existsSync(suggestionsDir)) {
     markdown(lines.join('\n'))
 }
 
+if (danger.git.created_files.some(f => path.basename(f) === ".editorconfig")) {
+    fail("A nested .editorconfig file may not be added to a package on DefinitelyTyped. Please respect the root .editorconfig.", danger.git.created_files.find(f => path.basename(f) === ".editorconfig"), 1)
+}
+
+
+for (const filename of danger.git.modified_files.concat(danger.git.created_files)) {
+    danger.git.diffForFile(filename).then(d => {
+        if (d.added.indexOf("\t") > -1) { // This is a dumb check for tabs, in lieu of running `prettier` on the diff'd parts
+            fail("The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.", filename, d.after.split("\n").findIndex(e => e.indexOf("\t") > -1) + 1)
+        }
+    })
```

Comment:
@orta I'm nor familiar with async code in a dangerfile - do I need to hook up a `.catch` on these callbacks, or do I not? Is it fine as-is? Would we prefer an `async` function we call (and then do we need to `.catch` on that)?

Reply:
Nah, this is fine - Danger's execution path is roughly that when the synchronous evaluation runs, then it hooks into the process exit code to catch all loose promises

Reply:
Oh yeah, the original point was about catching - Danger has a root catch which will fail the build and post a message with the stack, IMO, this shouldn't ever catch but it's handled at runtime level